src/tests/interpolation/test_interpolation_cubedsphere.cc

```diff
@@ -95,6 +69,66 @@ double dotProd(const Field& a, const Field& b) {
     return prod;
 }
 
+CASE("temporary const lvalue reference test") {
+
+    const auto grid = CubedSphereInterpolationFixture{}.sourceGrid_;
+
+    auto iterator = grid.lonlat().begin();
+    const auto begin = iterator;
+    for (idx_t i = 0; i < grid.size(); ++i) {
+
+        // This does not dangle, because C++ makes no sense!
+        const auto& lonLat = *(begin + i);
+        const auto refLonLat = *iterator;
+        ++iterator;
+
+        EXPECT_APPROX_EQ(lonLat.lon(), refLonLat.lon(), 1e-14);
+        EXPECT_APPROX_EQ(lonLat.lat(), refLonLat.lat(), 1e-14);
+
+        // Only now, *(begin + i), do you have my permission to die.
+    }
```

Comment:
So, funny thing, const lvalue refs to temporary object members can extend the lifetime of the host object...

Reply:
This seems like the worst code practice and should not be relied on for sure :)